cloudant-sync-datastore-core/src/main/java/com/cloudant/sync/internal/query/IndexCreator.java

```diff
@@ -394,4 +389,10 @@ private static String generateIndexName(Set<String> existingIndexNames) throws E
         }
     }
 
+    private static class ListIndexesCallable implements SQLCallable<List<Index>> {
```

Comment:
It saves the overhead of having the reference to the enclosing class, not sure how much of a overhead it is, @ricellis could probably explain more.